src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover_state.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import _ from 'lodash';
+import { createHashHistory, History } from 'history';
+import {
+  createStateContainer,
+  createKbnUrlStateStorage,
+  syncState,
+  ReduxLikeStateContainer,
+  IKbnUrlStateStorage,
+} from '../../../../../../../plugins/kibana_utils/public';
+import { esFilters, Filter, Query } from '../../../../../../../plugins/data/public';
+
+interface AppState {
+  /**
+   * Columns displayed in the table
+   */
+  columns?: string[];
+  /**
+   * Array of applied filters
+   */
+  filters?: Filter[];
+  /**
+   * id of the used index pattern
+   */
+  index?: string;
+  /**
+   * Used interval of the histogram
+   */
+  interval?: string;
+  /**
+   * Lucence or KQL query
+   */
+  query?: Query;
+  /**
+   * Array of the used sorting [[field,direction],...]
+   */
+  sort?: string[][];
+}
+
+interface GlobalState {
```

Comment:
Probably `GlobalState` could be completely removed from this file

Reply:
thx, done!